### PR TITLE
feature(calendar): Add maxdate

### PR DESCRIPTION
**Pull request by cwklausing**
* Adds maxDate param
* prevents click/state update on dates over maxDate
* adds date-disabled styling for dates over maxDate

### CODE REVIEW

File: src/Calendar.js [hunk 1]
```diff
@@ -46,8 +49,16 @@ export default class Calendar extends Component {
     const daysInMonth = getDaysInMonth(this.state.date) + 1
 
     for (let dateIndex = 1; dateIndex < daysInMonth; dateIndex += 1) {
-      const isDisabled = isBefore(new Date(format(this.state.date, `MM/${dateIndex}/YYYY`)), new Date(format(this.props.minDate, 'MM/DD/YYYY')))
-
+      const isDisabled =
+        isBefore(
+          new Date(format(this.state.date, `MM/${dateIndex}/YYYY`)),
```
Comment (tadjohnston) [inline src/Calendar.js]: I don't think we need to create the same date object multiple times. How about a const?
```
const dt = new Date(format(this.state.date, `MM/${dateIndex}/YYYY`)),

const isDisabled =
        isBefore(dt, new Date(format(this.props.minDate, 'MM/DD/YYYY'))) 
        ||
        isAfter(dt, new Date(format(this.props.maxDate, 'MM/DD/YYYY')))
``

Comment (tadjohnston) [reply src/Calendar.js]: Also, will this work if `minDate` or `maxDate` isn't defined?

Comment (cwklausing) [reply src/Calendar.js]: I'll update the const, but yeah, isAfter handles null cases, so it still works. I tested on the UI as well.

**Comment by lfender6445** [comment]
@cwklausing please update react-ui and rent-js to use the new datepicker

**Comment by lfender6445** [comment]
👍

File: src/Calendar.js [hunk 2]
```diff
@@ -44,10 +47,14 @@ export default class Calendar extends Component {
...
     const daysInMonth = getDaysInMonth(this.state.date) + 1
-
+    
     for (let dateIndex = 1; dateIndex < daysInMonth; dateIndex += 1) {
-      const isDisabled = isBefore(new Date(format(this.state.date, `MM/${dateIndex}/YYYY`)), new Date(format(this.props.minDate, 'MM/DD/YYYY')))
-
+      const dt = new Date(format(this.state.date, `MM/${dateIndex}/YYYY`))
+      const isDisabled =
+        isBefore(dt, new Date(format(this.props.minDate, 'MM/DD/YYYY'))) 
+        ||
```
Comment (orionstein) [inline src/Calendar.js]: Can you put this at the end of line 54

Comment (cwklausing) [reply src/Calendar.js]: certainly

**Pull request merged**